From ae83d063104f7e2c8e2abd84af4ddda24c49d45a Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Tue, 1 Sep 2026 13:38:38 +0900
Subject: [PATCH] build: Remove clang options unsupported by system clang

The Clang supplied by the scarthgap toolchain does not understand some
options emitted by Chromium's newer GN files.  Do not pass them when using
the system compiler.

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 build/config/compiler/BUILD.gn         | 9 ---------
 build/config/sanitizers/sanitizers.gni | 1 -
 2 files changed, 10 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 0dc7885fd8..39fce42d83 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -576,12 +576,6 @@ config("compiler") {
   if (is_clang) {
     # Flags for diagnostics.
     cflags += [ "-fcolor-diagnostics" ]
-    if (!is_win) {
-      cflags += [ "-fdiagnostics-show-inlining-chain" ]
-    } else {
-      # Combine after https://github.com/llvm/llvm-project/pull/192241
-      cflags += [ "/clang:-fdiagnostics-show-inlining-chain" ]
-    }
     if (diagnostics_print_source_range_info && !is_win) {
       cflags += [ "-fdiagnostics-print-source-range-info" ]
     }
@@ -615,9 +609,6 @@ config("compiler") {
     # The performance improvement does not seem worth the risk. See
     # https://crbug.com/484082200 for background and https://crrev.com/c/7593035
     # for discussion.
-    if (!is_wasm) {
-      cflags += [ "-fno-lifetime-dse" ]
-    }
 
     # TODO(hans): Remove this once Clang generates better optimized debug info
     # by default. https://crbug.com/765793
diff --git a/build/config/sanitizers/sanitizers.gni b/build/config/sanitizers/sanitizers.gni
index 04d86c5666..3693afe269 100644
--- a/build/config/sanitizers/sanitizers.gni
+++ b/build/config/sanitizers/sanitizers.gni
@@ -541,7 +541,6 @@ template("ubsan_hardening") {
         # be usable even in release builds, i.e. as widely as possible.
         # It's important not to have full-on UBSan workarounds activate
         # just because we built support for a specific sanitizer.
-        "-fsanitize-ignore-for-ubsan-feature=${invoker.sanitizer}",
       ]
       if (defined(invoker.cflags)) {
         cflags += invoker.cflags
